File: lib/cfr/builtins/libs/flutter/rendering/relativeRect.dart
import 'dart:core' as _fac9;
import 'dart:ui' as _a643;

import 'package:flutter/src/rendering/stack.dart' as _b1b6;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedRelativeRect extends _36c2.VMManagedBox<_b1b6.RelativeRect> {VMManagedRelativeRect({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) { table  [
'left'
] = vmObject.left;
table  [
'top'
] = vmObject.top;
table  [
'right'
] = vmObject.right;
table  [
'bottom'
] = vmObject.bottom;
table  [
'getHasInsets'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.hasInsets,]; } );
table  [
'shift'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_b1b6.RelativeRect>(object: vmObject.shift(_36c2.maybeUnBoxAndBuildArgument<_a643.Offset, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'inflate'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_b1b6.RelativeRect>(object: vmObject.inflate(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'deflate'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_b1b6.RelativeRect>(object: vmObject.deflate(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'intersect'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_b1b6.RelativeRect>(object: vmObject.intersect(_36c2.maybeUnBoxAndBuildArgument<_b1b6.RelativeRect, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'toRect'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Rect>(object: vmObject.toRect(_36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'toSize'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Size>(object: vmObject.toSize(_36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getHashCode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.hashCode,]; } );
table  [
'toString'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toString(),]; } ); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _b1b6.RelativeRect vmObject;

 }

void loadRelativeRect({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { 
table  [
'relativeRectFromSize'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return [_36c2.maybeBoxObject<_b1b6.RelativeRect>(object: _b1b6.RelativeRect.fromSize(_36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'relativeRectFromRect'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return [_36c2.maybeBoxObject<_b1b6.RelativeRect>(object: _b1b6.RelativeRect.fromRect(_36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'relativeRectLerp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 

final returnValue = _b1b6.RelativeRect.lerp(_36c2.maybeUnBoxAndBuildArgument<_b1b6.RelativeRect?, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_b1b6.RelativeRect?, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState), luaCallerArguments  [
3
]?.toDouble());if(returnValue!= null){return [_36c2.maybeBoxObject<_b1b6.RelativeRect?>(object: returnValue, hydroState: hydroState, table: _36c2.HydroTable()),];}return []; } );
_36c2.registerBoxer<_b1b6.RelativeRect>(boxer: ({required _b1b6.RelativeRect vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedRelativeRect(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 

File: runtime/flutter/rendering/relativeRect.ts

import { IRect } from "../../dart/ui/rect";
import { ISize } from "../../dart/ui/size";
import { IOffset } from "../../dart/ui/offset";
declare const flutter: {
rendering: {
relativeRectFromSize: (rect: IRect, container: ISize) => IRelativeRect;
relativeRectFromRect: (rect: IRect, container: IRect) => IRelativeRect;
relativeRectLerp: (a: IRelativeRect | undefined, b: IRelativeRect | undefined, t: number) => IRelativeRect | undefined
}
};
export interface IRelativeRect


{
left: number;
top: number;
right: number;
bottom: number;
getHasInsets: () => boolean;
shift: (offset: IOffset) => IRelativeRect;
inflate: (delta: number) => IRelativeRect;
deflate: (delta: number) => IRelativeRect;
intersect: (other: IRelativeRect) => IRelativeRect;
toRect: (container: IRect) => IRect;
toSize: (container: ISize) => ISize;
getHashCode: () => number;
toString: () => string;
}export class RelativeRect


{    public static fill = RelativeRect.fromLTRB(0.0, 0.0, 0.0, 0.0);
    public readonly left: number = undefined as any;
    public readonly top: number = undefined as any;
    public readonly right: number = undefined as any;
    public readonly bottom: number = undefined as any;
public static fromSize(rect: IRect, container: ISize) : IRelativeRect{
return flutter.rendering.relativeRectFromSize(rect, container);
}public static fromRect(rect: IRect, container: IRect) : IRelativeRect{
return flutter.rendering.relativeRectFromRect(rect, container);
}public static lerp(a: IRelativeRect | undefined, b: IRelativeRect | undefined, t: number) : IRelativeRect | undefined{
return flutter.rendering.relativeRectLerp(a, b, t);
}    private readonly _dart_getHasInsets: () => boolean = undefined as any;
    private readonly _dart_shift: (offset: IOffset) => IRelativeRect = undefined as any;
    private readonly _dart_inflate: (delta: number) => IRelativeRect = undefined as any;
    private readonly _dart_deflate: (delta: number) => IRelativeRect = undefined as any;
    private readonly _dart_intersect: (other: IRelativeRect) => IRelativeRect = undefined as any;
    private readonly _dart_toRect: (container: IRect) => IRect = undefined as any;
    private readonly _dart_toSize: (container: ISize) => ISize = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
public getHasInsets() : boolean {
    return this._dart_getHasInsets();
}
public shift(offset: IOffset) : IRelativeRect {
    return this._dart_shift(offset);
}
public inflate(delta: number) : IRelativeRect {
    return this._dart_inflate(delta);
}
public deflate(delta: number) : IRelativeRect {
    return this._dart_deflate(delta);
}
public intersect(other: IRelativeRect) : IRelativeRect {
    return this._dart_intersect(other);
}
public toRect(container: IRect) : IRect {
    return this._dart_toRect(container);
}
public toSize(container: ISize) : ISize {
    return this._dart_toSize(container);
}
public getHashCode() : number {
    return this._dart_getHashCode();
}
public toString() : string {
    return this._dart_toString();
}
}
